server/http4s-server/src/main/scala/sttp/tapir/server/http4s/Http4sServerInterpreter.scala

```diff
@@ -4,16 +4,25 @@ import cats.arrow.FunctionK
 import cats.effect.{Async, Sync}
 import cats.~>
 import org.http4s._
+import org.http4s.server.websocket.WebSocketBuilder2
 import sttp.capabilities.WebSockets
 import sttp.capabilities.fs2.Fs2Streams
 import sttp.tapir.server.ServerEndpoint
 
 trait Http4sServerInterpreter[F[_]] extends Http4sServerToHttpInterpreter[F, F] {
-  def toRoutes(se: ServerEndpoint[Fs2Streams[F] with WebSockets, F]): HttpRoutes[F] = toRoutes(List(se))
+  def toRoutes(se: ServerEndpoint[Fs2Streams[F] with WebSockets, F]): HttpRoutes[F] =
+    toRoutes(List(se))
 
-  def toRoutes(serverEndpoints: List[ServerEndpoint[Fs2Streams[F] with WebSockets, F]]): HttpRoutes[F] = {
-    toHttp(serverEndpoints)(fToG)(gToF)
-  }
+  def toRoutes(serverEndpoints: List[ServerEndpoint[Fs2Streams[F] with WebSockets, F]]): HttpRoutes[F] =
+    toHttp(serverEndpoints, webSocketBuilder = None)(fToG)(gToF)
+
+  def toRoutesWithWebSockets(se: ServerEndpoint[Fs2Streams[F] with WebSockets, F]): WebSocketBuilder2[F] => HttpRoutes[F] =
```

Comment:
From what i understand till now all methods supported WebSockets out of the box.
Maybe instead of creating new methods just add `webSocketBuilder: Option[WebSocketBuilder2[F]]` to current methods definitions.

Reply:
Well the problem is that now http4s makes the distinction between WS-using routes and "normal" ones. So I think we should respect this distinction in the interpreter, if that's how the base library works. Hopefully we can capture this distinction via types (using appropriate capabilities, see above).

I'd change the name, though, maybe `toWebSocketRoutes`?